mlflow/utils/rest_utils.py

```diff
@@ -317,44 +225,6 @@ def call_endpoints(host_creds, endpoints, json_body, response_proto):
                 raise e
 
 
-def cloud_storage_http_request(
```

Comment:
Is there a reason we need to move the functions in this file?

Reply:
It's duplicated in [download_cloud_file_chunk.py](https://github.com/mlflow/mlflow/pull/8352#diff-04f0e3acc70248c8d9a3ecf6f32d25c4ee926d2f2956cc0c74c32c1e6f7a8b33) So I removed all the duplicate functions in rest_utils.

Reply:
https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly might help avoid importing mlflow. For example, we could split `mlflow/utils/rest_utils.py` into two, one that import mlflow, one that doesn't import mlflow, and load the latter one when downloading artifact.